patroni/postgresql.py

```diff
@@ -1614,23 +1616,27 @@ def remove_data_directory(self):
             logger.exception('Could not remove data directory %s', self._data_dir)
             self.move_data_directory()
 
-    def basebackup(self, conn_url, env):
+    def basebackup(self, conn_url, env, options):
         # creates a replica data dir using pg_basebackup.
         # this is the default, built-in create_replica_method
         # tries twice, then returns failure (as 1)
         # uses "stream" as the xlog-method to avoid sync issues
+        # supports additional user-supplied options, those are not validated
         maxfailures = 2
         ret = 1
+        user_options = []
+        for k, v in options.items():
+            if v != '':
```

Comment:
What if `v is None?`

We can also do a list comprehension here:
```python
        user_options = ['--{0}={1}'.format(k, v) for k, v in options.items() if v]
```

Reply:
Yeah, good point on that as well. I actually wanted to change that anyway in order to support v being a list of options (to handle the `tablespace-mapping` option properly)